Core/src/ExponentialBackoff.php

```diff
@@ -136,6 +109,14 @@ public function execute(callable $function, array $arguments = [])
 
                 $delayFunction($calcDelayFunction($retryAttempt));
                 $retryAttempt++;
+                if ($this->retryListener) {
```

Comment:
Shouldn't the position of this function and the old onRetryExceptionFunction be same? Is this intentionally brought down?

Reply:
As mentioned [here](https://github.com/googleapis/google-cloud-php/pull/5637#discussion_r1161170652) and owing to the fact that adding/updating headers would be useless if we decide not to retry, it would make more sense to have this after deciding if we should retry again or not.